### PR TITLE
Add links to latest versions in release list

### PR DESCRIPTION
**Pull request by fhemberger**
Thought about this while reading #702. What do you think?

### CODE REVIEW

**Comment by lpinca** [comment]
Screenshots for the lazies:

**Before:**

before

**After:**

after

File: layouts/css/layout/_lists.styl [hunk 1]
```diff
@@ -4,6 +4,7 @@
 
   li
     display inline-block
+    padding-right .1em
```
Comment (lpinca) [inline layouts/css/layout/_lists.styl]: This also affects the top navigation.

File: locale/en/download/releases.md [hunk 2]
```diff
@@ -1,7 +1,20 @@
...
 ---
+
+### io.js &amp; Node.js
+Releases 1.x through 3.x were called "io.js" as they were part of the io.js fork. As of Node.js 4.0.0 the former release lines of io.js converged with Node.js 0.12.x into unified Node.js releases.
+
+<div class="highlight-box">
+    Looking for latest release of a version branch?
+
+    <ul class="list-divider-pipe">
+        <li><a href="https://nodejs.org/dist/latest-v6.x/">Node.js 6.x</a></li><li><a href="https://nodejs.org/dist/latest-v5.x/">Node.js 5.x</a></li>
```
Comment (lpinca) [inline locale/en/download/releases.md]: If you add a new line here, after the first `<li>`, we can probably avoid to add the padding in `_lists.styl`.

**Comment by fhemberger** [comment]
@lpinca Thanks for the screenshots, fixed the rest.

**Comment by lpinca** [comment]
LGTM.

**Pull request merged**